pkgs/applications/office/libreoffice/generate-libreoffice-srcs.py

```diff
@@ -0,0 +1,268 @@
+"""
+Converts the LibreOffice `download.lst` file into a Nix expression.
+
+Requires an environment variable named `downloadList` identifying the path
+of the input file, and writes the result to stdout.
+
+todo - Ideally we would move as much as possible into derivation dependencies.
+"""
+import collections, itertools, json, re, sys, os
+
+def main():
+
+    print('[')
+
+    for x in get_packages():
+
+        name = x['tarball']
+        md5 = x['md5']
+        brief = 'true' if x['brief'] else 'false';
+        subdir = x.get('subdir')
+
+        print('{')
+
+        print(f'  name = "{name}";')
+        print(f'  md5 = "{md5}";')
+        print(f'  brief = {brief};')
+
+        if subdir is not None:
+            print(f'  subDir = "{subdir}";')
+
+        print('}')
+
+    print(']')
+
+
+def get_packages():
+    """
+    All of the package data: What's parsed from download.lst,
+    plus our additions.
+    """
+    return apply_additions(get_packages_from_download_list(),
+                           get_additions())
+
+
+def get_additions():
+    """
+    A mapping from package name (the all-caps identifiers used in
+    `download.lst`) to a dict of additional attributes to set on the package.
+    """
+    with open('./libreoffice-srcs-additions.json') as f:
+        return json.load(f)
+
+
+def apply_additions(xs, additions):
+    for x in xs:
+        yield {**x, **additions.get(x['name'], {})}
```

Comment:
I don't know, maybe if the problem you want to fix is low readability of the bash script to people with weak knowledge of bash, `**x` thing in Python could be worth a comment like «merge two hashtables into one» for people who don't write Python.